### DISCUSSION TITLE
How convert meshes position? (cannon -> three || three -> cannon)

### DISCUSSION THREAD

**Discussion by nedcrow** [pmndrs/use-cannon#69] [General]
I made the land tilted 45 degrees.
And gravity too 45 degrees.

I want see stacked boxes at center of ground.
When this time, how convert position of plane or ground?

this like a,
`const posY = useRef(  convertY(plane.position, y) );`
`const posZ = useRef(  convertZ(plane.position, z) );`

`return <>`
`.   <Box position ={[x,posYs, posZ]} />`
`</>`

sandbox:
https://codesandbox.io/s/r3f-cannon-instanced-physics-z1iwd?file=/src/index.js

**Comment by codynova** [comment]
Hi @nedrow, I'm having trouble understanding your question. I can see how you've set the gravity at 45 degrees, but I don't understand what you are trying to achieve with positions.

**Reply by nedcrow** [reply]
Hi @codynova , I want multiple drop box. but boxes should straight align.

example 45 degrees,
ground position = [0, 0, 0]
box & ground rotation = [Math.PI/4, 0, 0]
box1 position = [0, 0 + 1/root2, 1/root2];
box2 position = [0, 2/root2, 2/root2];

and 50 degree, 75 dgree.. so I was wondering if there is an easy way (or formula).
